Markdown/Default.sublime-keymap

```diff
@@ -1,27 +1,20 @@
 [
     // Auto-pair asterisks
-    { "keys": ["*"], "command": "insert_snippet", "args": {"contents": "*$0*"}, "context":
```

Comment:
There was definitely a problem with this binding that extended beyond just typing formulas, since for example trying to add emphasis on a word has already been typed i.e. `some |words` would also trigger it, resulting in `some *|*words`, which is not how quotation auto pairing works, for example.

I would say that if you intend to input a literal `*` you *should* be escaping it with a backslash, i.e. when writing a formula, but without the context fixes fro the first part of my comment, they should not be re-added as-is.

Reply:
Markdown is quite smart in general with regards to distinguish literal `*` from bold/italics markup and I don't think it is acceptable to force everybody to escape them also in situations it is not needed.

Also the first example `some |word` is not so clear. How do you want to know whether typing an asterisk in front of the word should wrap the whole word vs. adding an empty bold/italic char/word. It might easily be someone wants to type a bold letter like `some |ord`  => `some *w*ord`.

Reply:
> How do you want to know whether typing an asterisk in front of the word should wrap the whole word

By selecting the word when I want do do that. This would match the quotation character matching, which I think of as reasonable and have gotten used to.

Reply:
... which is still possible.

Reply:
I think we might be talking past each other.

My primary concern about removing the binding altogether was that the probably most common use case of `a word |` => `a word *|*` will not be possible anymore. However, I do believe not having the binding is better than the current situation on `master`, which showed the behavior I outlined in the first comment of this thread, which is terrible. Typing <kbd>*</kbd> with a selection is unrelated to this.

Reply:
I think we should not auto-pair `* _ ~` by default, because of many false positives. For built-in keys, it's better to lean on the side of fewer features with fewer false positives, rather than more features with more false positives, because it's easier to opt in than to opt out.